### ISSUE TITLE
关于分类

### ISSUE THREAD

**Issue by HanAccount** [cdb342/Codes-for-IJCAI-2022-Paper-Zero-Shot-Logit-Adjustment#3]
你好！代码里我发现在分类过程中对adjusted cross-entropy操作就是
把输出加上算出来的两个概率吗？
logits = logits + p_y + p_0_Y

**Comment by cdb342** [comment]
![2022-05-01 21-06-09 的屏幕截图](https://user-images.githubusercontent.com/61510717/166147323-ec7e47fa-c460-48a8-a79b-f59e8245e620.png)
这里是加上两个概率的log值，再经过softmax之后得到公式13的结果。

**Comment by HanAccount** [comment]
![微信截图_20220502092908](https://user-images.githubusercontent.com/87269265/166174181-1232e6f6-932d-4090-841d-703d692caaaf.png)
求的就是两个先验概率，然后在分类阶段模型将预测输出加上这两个先验概率的log值再和
真实标签求loss对吧？
还有一个问题就是对先验求log后不是会小于0吗？这个操作就是将预测输出减去这两个值来减轻
论文摘要中所提的偏置性和同质性吗？

**Comment by cdb342** [comment]
![2022-05-02 10-49-35 的屏幕截图](https://user-images.githubusercontent.com/61510717/166178683-4e32b55b-9ad5-4ded-9787-1ed589473235.png)
模型输出加上两个先验的log值再带入到cross-entropy中就得到了公式15的结果。
这和是否小于0没有关系，先验对损失的调整依赖于不同类别先验log值之间的差值，或者说公式15中做exp操作后不同类别先验之间的比值。
并不是减轻偏置性和同质性，而是减轻偏置性和同质性造成的问题。

**Comment by HanAccount** [comment]
在AWA2上 我跑了300轮精度H最多就只能到72.6 
如果要用到其他方法中就只需要计算两个先验概率的log值然后加到输出就行了吧？

**Comment by cdb342** [comment]
我认为在不同设备上相差0.2%属于正常情况。在其它数据集上呢，可以达到文章里的结果吗？
如果其他方法使用类似的分类方法的话可以这么做。seen-unseen prior是自己定义的超参数，在不同方法里需要调整一下。

**Comment by HanAccount** [comment]
都跑了300轮，差不多只有AwA2上差了点，在CUB跑到了68.76，SUN上跑到了43.58比论文还高
使用的设备是RTX A5000 

seen-unseen prior使用的超参数调整有什么技巧吗？我看ratio参数有1000的也有几十的

**Comment by cdb342** [comment]
按照文章理论分析的话，粗粒度数据集生成的不可见类样本偏差相对细粒度数据集更大一点，所以需要更大的比值进行调整。

**Comment by HanAccount** [comment]
好的！非常感谢你耐心的回复
